src/thmap.c

```diff
@@ -805,19 +838,21 @@ stage_mem_gc(thmap_t *thmap, uintptr_t addr, size_t len)
 	gc->addr = addr;
 	gc->len = len;
 retry:
-	head = atomic_load_explicit(&thmap->gc_list, memory_order_relaxed);
-	atomic_thread_fence(memory_order_acquire);
-	gc->next = head;
-
-	if (!atomic_compare_exchange_weak(&thmap->gc_list, head, gc)) {
+	head = atomic_load_relaxed(&thmap->gc_list);
+	gc->next = head;	/* not yet published */
+	/* Release to subsequent acquire in thmap_stage_gc. */
+	if (!atomic_compare_exchange_weak_explicit(&thmap->gc_list, &head, gc,
+		memory_order_release, memory_order_relaxed)) {
```

Comment:
Tab + 4 spaces

Reply:
Usually in KNF, if it's one ( level, then it's four spaces, but each extra level of ( nesting warrants an additional indentation:

```
if (very_long_condition() &&
    another_very_long_condition)
```

vs

```
if (very_long_condition(with_nested,
	very_long, arguments) &&
    another_very_long_condition(with, nested,
	also_long, arguments))
```

This is the latter case.  Are you sure you prefer the non-KNF, 4-space version?

Reply:
Yes, I prefer the former. Most of the NetBSD code I've seen is also using such style and I'm not sure which part of KNF suggests the latter (but some developers do use the latter style too).